### PR TITLE
Port to firefox

### PR DESCRIPTION
**Pull request by dennisroche**
Great extension. Ported it across to Firefox. 

Extensions are compatible with Firefox, so just needed to add build/packing scripts.

### CODE REVIEW

**Comment by dennisroche** [comment]
Might need to look at https://github.com/LinusU/wext-shipit to manage the deployment to extension stores.

**Comment by akinjide** [comment]
> Great extension. Ported it across to Firefox.
> 
> Extensions are compatible with Firefox, so just needed to add build/packing scripts.

Great, let's ship it to Firefox. What's the LOE?

**Comment by akinjide** [comment]
> Might need to look at https://github.com/LinusU/wext-shipit to manage the deployment to extension stores.

I will review the link but I see you configured `AMO_JWT_ISSUER` and `AMO_JWT_SECRET`, how do we move this to Travis secure env or console env?

**Comment by dennisroche** [comment]
How do you currently publish this extension for Chrome? 

`AMO_JWT_ISSUER` and `AMO_JWT_SECRET` are the tokens for Firefox (**A**ddons.**M**ozilla.**O**rg). 

You would need to register with Firefox Addons (https://addons.mozilla.org/en-US/firefox/) to publish to their store to generate your own tokens (https://addons.mozilla.org/en-US/developers/addon/api/key/).

Then encrypt the values before including them in the `.travis.yaml` file (https://docs.travis-ci.com/user/environment-variables/#encrypting-environment-variables)

**Comment by akinjide** [comment]
> How do you currently publish this extension for Chrome?
> 
> `AMO_JWT_ISSUER` and `AMO_JWT_SECRET` are the tokens for Firefox (**A**ddons.**M**ozilla.**O**rg).
> 
> You would need to register with Firefox Addons (https://addons.mozilla.org/en-US/firefox/) to publish to their store to generate your own tokens (https://addons.mozilla.org/en-US/developers/addon/api/key/).
> 
> Then encrypt the values before including them in the `.travis.yaml` file (https://docs.travis-ci.com/user/environment-variables/#encrypting-environment-variables)

right, I have Chrome secret stored in Travis CI Environment Variables. I will follow this setup instructions

File: package.json [hunk 1]
```diff
@@ -4,14 +4,16 @@
   "license": "MIT",
   "author": "Akinjide Bankole <r@akinjide.me>",
   "scripts": {
-    "lint": "xo && stylelint extension/*.css",
+    "lint": "xo && stylelint extension/*.css && web-ext lint",
```
Comment (akinjide) [inline package.json:7]: include flags:

```
--output=text // The type of output to generate when reporting on errors.
--warnings-as-errors // Treat warnings as errors by exiting non-zero for warnings.
```

File: package.json [hunk 2]
```diff
@@ -4,14 +4,16 @@
...
     "test": "npm run lint",
     "test:watch": "ava --verbose --watch",
-    "build": "webpack --mode=production",
+    "build": "webpack --mode=production && web-ext build --source-dir=./extension/build --artifacts-dir=./dist --overwrite-dest",
```
Comment (akinjide) [inline package.json:10]: Better to use `--artifacts-dir=./extension-artifacts `

File: package.json [hunk 3]
```diff
@@ -4,14 +4,16 @@
...
     "build:watch": "webpack --mode=development --watch --devtool=source-map",
     "semver": "VERSION=$npm_package_version; echo $VERSION; dot-json extension/manifest.json version $VERSION --indent 2",
-    "pre:deploy": "cd extension && webstore upload --source=build --auto-publish",
-    "deploy": "npm-run-all semver build pre:deploy --sequential --print-name"
+    "start:firefox": "web-ext run --source-dir ./extension/build/",
```
Comment (akinjide) [inline package.json:13]: `--source-dir=./extension/build"`

File: package.json [hunk 4]
```diff
@@ -4,14 +4,16 @@
...
+    "deploy:firefox": "from-env web-ext sign --api-key %AMO_JWT_ISSUER --api-secret %AMO_JWT_SECRET --source-dir=./extension/build --artifacts-dir=./dist --channel=listed",
```
Comment (akinjide) [inline package.json:14]: remove flags and package `from-env`: 
```
--api-key %AMO_JWT_ISSUER  // using $WEB_EXT_API_KEY
--api-secret %AMO_JWT_SECRET  // using $WEB_EXT_API_SECRET
```

```
--artifacts-dir=./extension-artifacts
```

File: package.json [hunk 5]
```diff
@@ -4,14 +4,16 @@
...
+    "deploy:chrome": "cd extension && webstore upload --source=build --auto-publish",
+    "deploy": "npm-run-all semver build deploy:chrome deploy:firefox --sequential --print-name"
```
Comment (akinjide) [inline package.json:16]: run chrome and firefox deploy in parallel `npm-run-all semver build --print-name --parallel deploy:*`

**Comment by dennisroche** [comment]
Build is failing with `web-ext lint` warnings (as we turned on `--warnings-as-errors`)

```
UNSAFE_VAR_ASSIGNMENT   Unsafe assignment to innerHTML   Due to both security and performance concerns, this may not be set using dynamic values which have not been   jquery.slim.min.js   2      10644
                                                         adequately sanitized. This can lead to security issues or fairly serious performance degradation.
```

**Comment by akinjide** [comment]
Anyway, to suppress warning or skip dependencies?

File: package.json [hunk 6]
```diff
@@ -29,10 +32,12 @@
     "copy-webpack-plugin": "^5.1.1",
     "dot-json": "^1.0.4",
     "faker": "^4.1.0",
+    "from-env": "^1.1.4",
```
Comment (akinjide) [inline package.json:35]: line 14, 35: Please remove from-env dependencies.

File: package.json [hunk 7]
```diff
@@ -4,18 +4,21 @@
...
-    "deploy": "npm-run-all semver build pre:deploy --sequential --print-name"
+    "start:firefox": "web-ext run --source-dir ./extension/build",
+    "deploy:firefox": "from-env web-ext sign --source-dir=./extension/build --artifacts-dir=./extension-artifacts --channel=listed",
+    "deploy:chrome": "cd extension && webstore upload --source=build --auto-publish",
+    "deploy": "npm-run-all semver build --parallel deploy:* --sequential --print-name"
   },
   "dependencies": {
     "element-ready": "^3.0.0",
     "jquery": "^3.5.1",
+    "uri-js": "^4.2.2",
```
Comment (akinjide) [inline package.json:21]: Is `uri-js` currently in use?

File: package.json (hunk 6 above)
Comment (dennisroche) [reply package.json]: removed

File: package.json (hunk 7 above)
Comment (dennisroche) [reply package.json]: removed

**Comment by dennisroche** [comment]
> Anyway, to suppress warning or skip dependencies with `--ignore-files` on web-ext lint?

Doesn't appear a way to ignore files, see https://extensionworkshop.com/documentation/develop/web-ext-command-reference/#web-ext_lint despite being discussed in https://github.com/mozilla/web-ext/issues/131

Need to remove `--warnings-as-errors` to get this to pass CI. 😢

**Comment by akinjide** [comment]
> > Anyway, to suppress warning or skip dependencies with `--ignore-files` on web-ext lint?
> 
> Doesn't appear a way to ignore files, see https://extensionworkshop.com/documentation/develop/web-ext-command-reference/#web-ext_lint despite being discussed in [mozilla/web-ext#131](https://github.com/mozilla/web-ext/issues/131)
> 
> Need to remove `--warnings-as-errors` to get this to pass CI. 😢

Let's remove it and add later when necessary

**Comment by dennisroche** [comment]
> Test fails https://travis-ci.org/github/akinjide/refined-aws/builds/699670729

sorry about that; i didn't run tests before pushing. 

build fixed https://travis-ci.org/github/akinjide/refined-aws/builds/699945738

**Pull request merged**